cpp/src/arrow/filesystem/localfs.h

```diff
@@ -34,10 +34,24 @@ namespace fs {
 
 /// Options for the LocalFileSystem implementation.
 struct ARROW_EXPORT LocalFileSystemOptions {
+  static constexpr uint32_t kDefaultDirectoryReadahead = 1u;
+  static constexpr uint32_t kDefaultBatchSize = 1000u;
```

Comment:
Style nit: let's avoid gratuitous use of unsigned integers. Can make these `int` or `int32_t`.

Reply:
Just curious: why are `uint32_t`:s bad?

Reply:
They are not bad per se, but the general tendency should be to use them only when necessary (for example you want to perform unsigned computations, or are you reading data provided by a third-party API), as otherwise you inevitably end up mixing signed and unsigned which is generally annoying.